test/adding_tests.md

```diff
@@ -192,38 +208,43 @@ err := test.CheckConfigurationState(clients.ServingClient, configName, func(c *v
 })
 ```
 
-_See [crd_checks.go](./crd_checks.go) and
+_v1alpha1testing is alias for package `knative.dev/serving/pkg/testing/v1alpha1`_
+
+_For knative crd state, for example `Config`. You can see the code in [configuration.go](./v1alpha1/configuration.go).
+For kubernetes objects see
 [kube_checks.go](https://github.com/knative/pkg/blob/master/test/kube_checks.go)._
 
 ### Verify resource state transitions
 
 To use the [check functions](#check-knative-serving-resources) you must provide
 a function to check the state. Some of the expected transition states (as
-defined in [the Knative Serving spec](../docs/spec/spec.md)) are expressed in
-functions in [states.go](./states.go).
+defined in [the Knative Serving spec](https://github.com/knative/docs/blob/master/docs/serving/spec/knative-api-specification-1.0.md))
```

Comment:
Why not a relative link here?

Reply:
The spec.md has been moved to the Knative Docs Repository.
So I replace it with url, I think it is convenient for user.

Reply:
The spec.md has been moved to the Knative Docs Repository.
So I replace it with url, I think it is convenient for user.

Reply:
Sorry, I misread the link. Thanks for clarifying.